diff --git a/src/core/ipv4/ip4_napt.c b/src/core/ipv4/ip4_napt.c
index c02b599e..c1d518d5 100644
--- a/src/core/ipv4/ip4_napt.c
+++ b/src/core/ipv4/ip4_napt.c
@@ -41,6 +41,10 @@
 
 #if ESP_LWIP && LWIP_IPV4 && IP_NAPT
 
+#ifdef ESP_IDF_LWIP_HOOK_FILENAME
+#include ESP_IDF_LWIP_HOOK_FILENAME
+#endif
+
 #include <assert.h>
 #include <stdbool.h>
 #include <string.h>
@@ -613,6 +617,11 @@ ip_napt_add(u8_t proto, u32_t src, u16_t sport, u32_t dest, u16_t dport, u32_t s
     return mport;
   }
   LWIP_DEBUGF(NAPT_DEBUG, ("ip_napt_add() failed to insert\n"));
+
+#ifdef IP_NAPT_ADD_FAILED_HOOK
+    IP_NAPT_ADD_FAILED_HOOK();
+#endif
+
   return 0;
 }
 
@@ -1075,4 +1084,45 @@ ip_napt_get_stats(struct stats_ip_napt *stats)
 }
 #endif /* LWIP_STATS */
 
+#define IP_GET_V4(ipaddr)          (ip_2_ip4(ipaddr))
+
+void ip_napt_table_clear(void)
+{
+    struct netif *each_netif;
+    u16_t i = 0, next = 0;
+    struct ip_napt_entry *t;
+
+    NETIF_FOREACH(each_netif) {
+        if (netif_is_up(each_netif)
+            && !each_netif->napt
+            && !ip_addr_isany(&each_netif->ip_addr)
+            && IP_IS_V4(&each_netif->ip_addr)
+            && !ip4_addr_isloopback(IP_GET_V4(&each_netif->ip_addr))) {
+            /* napt used on another interface, no need for cleanup */
+            return;
+        }
+    }
+
+    uint32_t count = 0;
+    for (i = napt_list; i != NO_IDX; i = next) {
+        t = NT(i);
+        next = t->next;
+        count++;
+    }
+    LWIP_DEBUGF(NAPT_DEBUG, ("before napt count=%"PRIu32"\r\n", count));
+
+    for (i = napt_list; i != NO_IDX; i = next) {
+        t = NT(i);
+        next = t->next;
+        ip_napt_free(t);
+    }
+
+    for (i = napt_list; i != NO_IDX; i = next) {
+        t = NT(i);
+        next = t->next;
+        count++;
+    }
+    LWIP_DEBUGF(NAPT_DEBUG, ("after napt count=%"PRIu32"\r\n", count));
+}
+
 #endif /* ESP_LWIP && LWIP_IPV4 && IP_NAPT */
