diff --git a/src/netif/ethernet.c b/src/netif/ethernet.c
index dd171e28..57d9cb9d 100644
--- a/src/netif/ethernet.c
+++ b/src/netif/ethernet.c
@@ -291,7 +291,7 @@ ethernet_output(struct netif * netif, struct pbuf * p,
   } else
 #endif /* ETHARP_SUPPORT_VLAN && defined(LWIP_HOOK_VLAN_SET) */
   {
-    if (pbuf_add_header(p, SIZEOF_ETH_HDR) != 0) {
+    if (pbuf_add_header_force(p, SIZEOF_ETH_HDR) != 0) {
       goto pbuf_header_failed;
     }
   }
